[BOJ 4779] 칸토어 집합 / S3

Eunjin3395 opened the pull request:
### 🚀 문제 번호
Resolve: {#9}  
<br/>

### 🔍 구현 방법
<!-- 문제를 해결한 구체적인 방법을 설명해주세요. -->
어제 풀었던 2447번 별찍기랑 되게 비슷해서 이번 문제도 비슷한 방법으로 풀 수 있었습니다..!
구간을 세등분해서 왼쪽, 오른쪽은 재귀 호출, 가운데는 N//3만큼 공백을 출력했습니다.
" - "은 N이 1일때만 출력했습니다.
```
# 입력 받기
array = []

while True:
    try:
        line = input()
        array.append(line)
    except EOFError:
        break


def recursion(N):
    # base case
    if(N == 1):
        print("-", end="")
        return

    recursion(N//3)
    print(" "*(N//3), end="")
    recursion(N//3)


for elem in array:
    N = int(elem)
    recursion(3**N)
    print()

```
<br/>

### ⭐️ 리뷰 Point
<!-- 이 PR에 대한 논의하고 싶은 사항이나 리뷰어에게 특별히 확인 요청하고 싶은 부분 등을 적어주세요. -->
파이썬 파일 끝까지 한번에 입력받고 개행 기준으로 나누는 방법이 또 다른 방법이 있을까요...?

rimi3226 commented:
파일 끝까지 읽는 방법을 저도 찾아봤는데요!
일단 강의에서 나왔던 방법은
```
# input 
while True:
	try:
		N = int(input())
		func(N,True)
		print("");
	except:
		break;
```
이 방법이었고, 
따로 또 찾아본 방법은 
```
import sys

lines = sys.stdin.readlines()
for line in lines:
    A, B = map(int, line.split())
    print(A+B)
```
```
while True:
    try:
        A, B = map(int, input().split())
        print(A+B)
    except EOFError:
        break
```
이거예요!
근데 첫번째로 해보니까 for문 안에서 map으로 split하고나니 int 값이 input에 들어가는게 아니라 map object 값이 들어가더라구요,,? 그래서 for문으로 다시 돌아서 찾아야했습니다. 그래서 저는 강의에 있는 방식이 제일 편하다고 느껴져요!

rimi3226 commented on RecursiveFunction/BOJ4779/eunjin.py:
```diff
@@ -0,0 +1,26 @@
+        return
+
+    recursion(N//3)
+    print(" "*(N//3), end="")
```
이렇게 나타내도 좋네요! 강의에서 본 방식은
```
	print(' ' * (3 ** (k-1)), end="")
``` 
이거였습니다. 3^(N-1)
같은 말인데 다른 방식으로 표현된거 보니까 재밌네요 ㅎㅎ

Eunjin3395 replied:
강의에서는 보니까 작은 인덱스에서 점점 커지는 방식 같더라구요..! 저는 완전 반대의 방식으로 했네요 ㅋㅋㅋ

rimi3226 replied:
헉 그러네요ㅋㅋㅋ 강의에서 여러 풀이를 알려줘서 좋은 것 같아요!

Eunjin3395 commented:
> 파일 끝까지 읽는 방법을 저도 찾아봤는데요! 일단 강의에서 나왔던 방법은
> 
> ```
> # input 
> while True:
> 	try:
> 		N = int(input())
> 		func(N,True)
> 		print("");
> 	except:
> 		break;
> ```
> 
> 이 방법이었고, 따로 또 찾아본 방법은
> 
> ```
> import sys
> 
> lines = sys.stdin.readlines()
> for line in lines:
>     A, B = map(int, line.split())
>     print(A+B)
> ```
> 
> ```
> while True:
>     try:
>         A, B = map(int, input().split())
>         print(A+B)
>     except EOFError:
>         break
> ```
> 
> 이거예요! 근데 첫번째로 해보니까 for문 안에서 map으로 split하고나니 int 값이 input에 들어가는게 아니라 map object 값이 들어가더라구요,,? 그래서 for문으로 다시 돌아서 찾아야했습니다. 그래서 저는 강의에 있는 방식이 제일 편하다고 느껴져요!

헉 그렇군요,,진짜 첫번째가 제일 간단해보이긴 하네요!! 저도 이제 저런 방식으로 해봐야겠어요 자세한 설명 감사합니다👍

hzee97 reviewed (approved):
`
    except EOFError:
        break
`
이렇게도 구현할 수 있네요!
